### PR TITLE
[Profiling] Add backwards-compatibility test

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we add a basic backwards-compatibility test that checks whether indices and data streams can be properly upgraded to the latest version.

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
I'm opening this as a draft PR for the moment because the BWC tests are effectively disabled before 8.11.0 because only then we'll have the flag `wait_for_resources_created` available in the profiling status check API (introduced in https://github.com/elastic/elasticsearch/pull/99655).

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine merge upstream

**Comment by danielmitterdorfer** [comment]
Marking as ready for review as the `8.11` branch has been cut.

File: rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "profiling.status":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/observability/current/universal-profiling.html",
+      "description":"Returns basic information about the status of Universal Profiling."
+    },
+    "stability":"stable",
+    "visibility":"private",
+    "headers":{
+      "accept": [ "application/json"]
```
Comment (rockdaboot) [inline rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json:10]: nit
```suggestion
      "accept": ["application/json"]
```

File: rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          "methods":[
+            "GET"
+          ]
+        }
+      ]
+    },
+    "params":{
+      "master_timeout":{
+        "type":"time",
+        "description":"Explicit operation timeout for connection to master node"
```
Comment (rockdaboot) [inline rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json:25]:
```suggestion
        "description":"Explicit operation timeout for connections to the master node"
```

**Comment by rockdaboot** [comment]
What is the intention of using the `150_` prefix?

This "priority-as-prefix" stems from *nix and relies on `ls` sorting the file names in the correct order. But this only works as expected with `00..99` because `ls` sorts the file names as strings and isn't aware of the priorities as numeric fields.

Maybe the reader(s) of the directory handle(s) this differently, but can we rely on this?

File: rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json (hunk 2 above)
Comment (danielmitterdorfer) [reply rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json:25]: Good point. However, for consistency reasons I like to keep it that way: this is common parameter in many API calls that is documented with this exact wording everywhere else and the user-facing documentation should be identical.

**Comment by danielmitterdorfer** [comment]
> What is the intention of using the 150_ prefix?

This is an existing convention for the these tests and I adopted it here.

File: rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json (hunk 2 above)
Comment (rockdaboot) [reply rest-api-spec/src/main/resources/rest-api-spec/api/profiling.status.json:25]: Hm, then maybe we should fix this everywhere, but not in this PR :)

**Comment by rockdaboot** [comment]
> > What is the intention of using the 150_ prefix?
> 
> This is an existing convention for the these tests and I adopted it here.

So it unclear where it stems from? Do you know of any code or documentation that I can read up about it. I am very interested in the details (one of my bad habits).

**Review by rockdaboot — APPROVED** [review]

**Comment by danielmitterdorfer** [comment]
> So it is unclear where it stems from? Do you know of any code or documentation that I can read up about it. I am very interested in the details (one of my bad habits).

I did some digging but PRs that touch such files are at least from early 2016. My guess is that back then somebody came up with the idea to visually sort them. I'm also not aware of any documentation around it. In fact, I am not aware of any documentation about backwards-compatibility tests except a very brief bit in https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc on how to execute them.

**Comment by danielmitterdorfer** [comment]
The test failure on `part-2` is likely fixed by https://github.com/elastic/elasticsearch/pull/100378 which has recently been merged to `main`.

**Comment by danielmitterdorfer** [comment]
@elasticmachine merge upstream

**Comment by danielmitterdorfer** [comment]
seems one PR check is hanging indefinitely.

@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by danielmitterdorfer** [comment]
build timed out (CI infrastructure issue)

@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by danielmitterdorfer** [comment]
build timed out (CI infrastructure issue)

@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by danielmitterdorfer** [comment]
build timed out again (CI infrastructure issue)

@elasticsearchmachine run elasticsearch-ci/part-1

**Comment by danielmitterdorfer** [comment]
@elasticmachine update branch

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine run elasticsearch-ci/bwc-snapshots

**Comment by danielmitterdorfer** [comment]
@elasticsearchmachine run elasticsearch-ci/7.17.14 / bwc-snapshots

**Pull request merged**